src/Audacia.Seed/Properties/SeedNavigationPropertyConfiguration.cs

```
@@ -0,0 +1,155 @@
+    {
+        ArgumentNullException.ThrowIfNull(getter);
+
+        if (getter.Body.Type != typeof(TNavigation))
+        {
+            return null;
+        }
+
+        if (getter.GetPropertyInfo() != Getter.GetPropertyInfo())
+        {
+            return null;
+        }
+
+        var seedToReturn = SeedConfiguration as EntitySeed<TNavigation>;
+        return seedToReturn;
+    }
+
+    /// <summary>
+    /// Gets a lambda to the property to populate.
+    /// </summary>
+    internal Expression<Func<TEntity, TNavigation?>> Getter { get; } = getter;
+
+    /// <summary>
+    /// Gets a list of seed configurations to use, in order in which they will be used.
+    /// </summary>
+    private IEntitySeed<TNavigation> SeedConfiguration { get; } = seedConfiguration;
+
+    /// <inheritdoc />
+    public void Apply(TEntity entity, ISeedableRepository repository, int index, TEntity? previous)
+    {
+        ArgumentNullException.ThrowIfNull(repository);
+        SeedConfiguration.Repository ??= repository;
+
+        var value = (previous != null
+                     && SeedConfiguration.Options.InsertionBehavior == SeedingInsertionBehaviour.TryFindExisting
+                        ? Getter.Compile().Invoke(previous)
+                        : null)
+                    ?? GetValueToSet(repository, index, previous, SeedConfiguration);
+
+        var obj = Getter.GetPropertyOwner(entity);
+
+        if (Getter.Body is MemberExpression memberSelectorExpression)
+        {
+            var property = (PropertyInfo)memberSelectorExpression.Member;
+            property.SetValue(obj, value, null);
+        }
+    }
+
+    private static TNavigation GetValueToSet(ISeedableRepository repository, int index, TEntity? previous,
+        IEntitySeed<TNavigation> navigationSeed)
+    {
+        TNavigation? value = null;
+        if ((navigationSeed.Options.InsertionBehavior == SeedingInsertionBehaviour.MustFindExisting ||
```

If we're using `MustFindExisting`, should we throw an exception if `value` is null?

---

Good idea, but don't think it's appropriate now that it's renamed to `TryFindExisting`